helena-project/tock
# Enable graphical debugging in VSCode

### Pull Request Overview
This adds functionality to perform graphical debugging in VSCode on nRF51-DK and nRF52-DK.
It should be no problem to port this to the other boards if anyone finds this useful
My intention is that this will make remote debugging easier directly in the editor but of course an instance of gdb_server must be created separately

OT: I'm pretty excited that this work

### Testing Strategy

1. Start gdb server
2. Open VSCode 
3. Open Folder ```path to tock```
4. Install the extension [Native Debug](https://github.com/WebFreak001/code-debug)
5. Open debug view
6. Select board to debug
7. Press run debugging
8. The following should show:

![vscode](https://user-images.githubusercontent.com/14157425/36278220-70bde20a-1293-11e8-86a0-26e8aeb63325.png)

9. Then it easy to enable further breakpoints and so on, by clicking left of the source line number

(* tested both on nRF51-DK and nRF52-DK)

### TODO or Help Wanted


### Documentation Updated

- [X] Kernel: The relevant files in `/docs` have been updated or no updates are required.
- [X] Userland: The application README has been added, updated, or no updates are required.

### Formatting

- ~~[ ] `make formatall` has been run.~~
